cirq/contrib/qasm_import/_parser_test.py

```diff
@@ -0,0 +1,153 @@
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pytest
+
+import cirq.testing as ct
+from cirq import Circuit
+from cirq.contrib.qasm_import import QasmException
+from cirq.contrib.qasm_import._parser import QasmParser
+
+
+def test_format_header_circuit():
+    parser = QasmParser("OPENQASM 2.0;")
+
+    parsed_qasm = parser.parse()
```

Comment:
Shouldn't the source go into the parse method, not the constructor?

Reply:
Same argument as in the lexer (above) - the initialization is setup in the constructor for a one-time use object. I'm happy to move it around if you feel strongly about it.